inc/saiport.h

```diff
@@ -1498,6 +1501,71 @@ typedef enum _sai_port_attr_t
      */
     SAI_PORT_ATTR_QOS_TC_AND_COLOR_TO_MPLS_EXP_MAP,
 
+    /**
+     * @brief Port Down Error Status
+     *
+     * @type sai_port_err_status_list_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_ERR_STATUS_LIST,
+
+    /**
+     * @brief Fabric port Attached
+     *
+     * Signifies if the fabric port is attached
+     *
+     * @type bool
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_FABRIC_ATTACHED,
+
+    /**
+     * @brief Attached Switch type.
+     *
+     * Signifies the attached switch type
+     *
+     * @type sai_switch_type_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_FABRIC_ATTACHED_SWITCH_TYPE,
+
+    /**
+     * @brief Attached Switch ID.
+     *
+     * Signifies the attached switch ID
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_FABRIC_ATTACHED_SWITCH_ID,
+
+    /**
+     * @brief Attached Port Index.
+     *
+     * Signifies the attached port index
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_FABRIC_ATTACHED_PORT_INDEX,
+
+    /**
+     * @brief Fabric port reachability
+     *
+     * @type sai_fabric_port_reachability_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_FABRIC_REACHABILITY,
+
+    /**
+     * @brief System port for the port
+     *
+     * @type sai_object_id_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_SYSTEM_PORT
+     */
+    SAI_PORT_ATTR_SYSTEM_PORT,
+
```

Comment:
I think it would be good if we can query - attached_core_index and attached_core_port_index using the port table for ports local to the asic. This will help an application call flow - where the application can query the port table for these two attributes and then use them when requesting the creation of the system_port.

Without this information being available from the port_table - the application would need additional knowledge upfront about which ports belong to which core. It would less error prone if instead it uses the information returned from SAI.

Reply:
Are you thinking of a future dynamic port breakout case where a new local port is created without an associated system port?  We can consider this as an enhancement.

The initial use case was to provide the local port and system port mapping upfront to create_switch using SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST.   If this info is not saved by the app,  you can retrieve the info from the local port list using get "SAI_PORT_ATTR_SYSTEM_PORT" attribute followed by get "SAI_SYSTEM_PORT_ATTR_CONFIG_INFO".

Reply:
This is not for dynamic port breakout. Today the configuration specifies the lanes that for each port. The configuration flow that I am looking is something line this.

1. create_switch (switch_id,max_cores) /* for DNX the ".bcm" specifies the initial ports that are created. System ports are not created here  */
2. query SAI for port table /* this will return attached_core_index and attached_core_port_index. */
3. create_system_port (system_port_id,attached_core_index,attached_core_port_index, ...) /* system port id is application assigned. The other two attributes were previously queried from SAI*/

If we do this - there would not be a need to embed attached_core_index" and "attached_core_port_index" in application configuration files (only lane information would be necessary). They would be read from the SAI port tables and used during system port creation.